exporter/otlpexporter/otlp.go

```diff
@@ -110,7 +111,7 @@ func (e *exporter) start(ctx context.Context, host component.Host) error {
 	if e.config.Arrow != nil && e.config.Arrow.Enabled {
 		ctx := e.enhanceContext(context.Background())
 
-		e.arrow = arrow.NewExporter(e.config.Arrow, e.settings.TelemetrySettings, e.clientConn, e.callOptions)
+		e.arrow = arrow.NewExporter(*e.config.Arrow, e.settings.TelemetrySettings, arrowpb.NewArrowStreamServiceClient(e.clientConn), e.callOptions)
```

Comment:
Thoughts on moving the client creation back to `arrow.NewExporter()` and creating a `arrow.newExporter()` that lets you override the client for tests? :)

Reply:
I considered this based on your suggestion. I think you mean to pass the `*grpc.ClientConn` in to the Exporter, then override it. That seems to create more trouble, since I have to pass a `nil` in the tests? Then I considered eliminating the `[]grpc.CallOption` parameter as well, which would abstract all of the gRPC service call from the `arrow/internal` object. Neither seemed as close to the literal thing I wanted to test as what's there now.

What are your thoughts about the approach taken here?

Reply:
(Happy to rework this as the code develops!)